ion-transformation-storage-api/src/main/java/com/connexta/transformation/storage/api/Status.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api;
+
+import java.time.Instant;
+
+public interface Status {
+
+  /**
+   * Returns true if the process is still in progress, otherwise, returns false.
+   *
+   * @return true if still in progress, otherwise false
+   */
+  boolean isInProgress();
+
+  /**
+   * Returns true if the process failed, otherwise, returns false.
+   *
+   * @return true if the process failed, otherwise false
+   */
+  boolean failed();
+
+  /**
+   * Returns the start time of the process.
+   *
+   * @return the time the process started
+   */
+  Instant getStartTime();
```

Comment:
❗️ Should also include the duration and the last modified timestamp (or creation time) as optionals